gap/congruences/conginv.gi

```diff
@@ -490,16 +490,16 @@ SEMIGROUPS.KernelTraceClosure := function(S, kernel, traceBlocks, pairstoapply)
   od;
   nrk := Length(kernelgenstoapply);
   Elements(kernel);
-  pairstoapply := List(pairstoapply, x -> [Position(idsdata, RightOne(x[1])),
-                                           Position(idsdata, RightOne(x[2]))]);
+  pairstoapply := List(pairstoapply, x -> [Position(idsmgp, RightOne(x[1])),
+                                           Position(idsmgp, RightOne(x[2]))]);
   nr := Length(pairstoapply);
 
   # Calculate traceUF from traceBlocks
   traceUF := UF_NEW(Length(idslist));
   for i in [1 .. Length(traceBlocks)] do
-    pos1 := Position(idsdata, traceBlocks[i][1]);
+    pos1 := Position(idsmgp, traceBlocks[i][1]);
     for j in [2 .. Length(traceBlocks[i])] do
```

Comment:
`Position(S, x)` only checks if `x` is known to belong to `S` or not, it does not perform any enumeration of the semigroup. For example,

    gap> S := RandomSemigroup();
    <semigroup of 4x4 max-plus matrices with 8 generators>
    gap> Position(S, S.1);
    fail

It would be safer to use `PositionCanonical` here which will enumerate the semigroup until the position of the element is determined or the semigroup is fully enumerated and the element is known to not belong to it.